### PR TITLE
Doc/improve

### PR DESCRIPTION
**Pull request by calebmer**
I rendered the docs site and wanted to improve some things. Thus, this PR.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -70,7 +86,17 @@ impl Error {
     self.hint.to_owned()
   }
 
-  /// Convenience function for saying a thing failed validation.
+  /// Convenience function for saying a thing failed validation using
+  /// `ErrorCode::BadRequest`.
+  ///
+  /// # Example
+  /// ```rust
```
Comment (svmnotn) [inline src/error.rs:93]: Isn't the default here to auto assume rust?

Comment (calebmer) [inline src/error.rs:93]: I guess, but I prefer being explicit in the case where we need to repurpose the docs. We can copy/paste to a new markdown file more easily. Also for the case where we might not have rust code in a preformatted block.

**Comment by svmnotn** [comment]
ok, it looks to be fine.  :+1: merge

**Comment by calebmer** [comment]
Tests are failing because of out of sync dependencies.

**Pull request merged**